### PR TITLE
Refactor simple_router 

### PR DESCRIPTION
**Pull request by Drake-Eidukas**
This makes our `Router` code cleaner / more modular, and also slightly improves test time (~6-10 seconds in single-execution mode). This is going to facilitate implementing batched share opening.

### CODE REVIEW

File: honeybadgermpc/router.py [hunk 1]
```diff
@@ -1,35 +1,67 @@
...
+
+
+class SimpleRouter(Router):
+    def __init__(self, num_parties):
+        super().__init__(num_parties)
+
+        # Mailboxes for each party
+        self._queues = [asyncio.Queue() for _ in range(num_parties)]
+
+    async def _recv(self, player_id: int) -> object:
```
Comment (smkuls) [inline honeybadgermpc/router.py]: why the `_` prefix? These are supposed to be public methods.

File: tests/test_simple_router.py [hunk 2]
```diff
@@ -17,7 +17,7 @@
 
 
 @mark.asyncio
-async def test_simple_router(test_router):
+async def test_test_router(test_router):
```
Comment (smkuls) [inline tests/test_simple_router.py]: nit: `test_router`

**Review by smkuls — APPROVED** [review]
Looks good to me, except the `._` prefix in the name which I don't think should be there. `_` prefix afaik is to denote private methods, but these are public methods.

File: honeybadgermpc/router.py (hunk 1 above)
Comment (Drake-Eidukas) [reply honeybadgermpc/router.py]: At the time, the intention was to access these via the `.recvs` and `.sends` attributes, but I agree now.

**Pull request merged**